explainer.md

```diff
@@ -405,20 +393,67 @@ function beginInlineXRSession() {
+
+Features may be unavailable for a number of reasons, among which is the fact not all devices which support WebXR can support the full set of features. Another consideration is that some features expose [sensitive information](privacy-security-explainer.md#sensitive-information) which may require a clear signal of [user intent](privacy-security-explainer.md#user-intent) before functioning. Any feature which requires this signal to be provided via [explicit consent](privacy-security-explainer.md#explicit-consent) must request this consent prior to the session being created. This ensures a consistent experience across all hardware form-factors, regardless of whether the UA has a [trusted immersive UI](privacy-security-explainer.md#trusted-immersive-ui) available.
+
+WebXR allows the following features to be requested:
+* `local`
+* `local-floor`
+* `bounded-floor`
+* `unbounded`
+
+This list is currently limited to a subset of reference space types, but in the future will expand to include additional features. Some potential future features under discussion that would be candidates for this list are: eye tracking, plane detection, geo alignment, etc.
+
+Developers communicate their feature requirements by categorizing them into one of the following groups:
+* **Required** This feature must be available in order for the experience to function at all. If [explicit consent](privacy-security-explainer.md#explicit-consent) is necessary, users will be prompted in response to `xr.requestSession()`. Session creation will be rejected if the feature is unavailable for the XR device or if the UA determines the user does not wish the feature enabled.
+* **Optional** The experience would like to use this feature for the entire session, but can function without it. Again, if [explicit consent](privacy-security-explainer.md#explicit-consent) is necessary, users will be prompted in response to `xr.requestSession()`. However, session creation will succeed regardless of the feature's hardware support or user intent. Developers must not assume optional features are available in the session and check the result from attempting to use them.
+
+If the UA does not recognize an entry in `XRSession.requiredFeatures` the session will not be created. Under some circumstances, features recognized by the UA but not explicitly listed in these arrays may be implicitly added to `XRSessionInit.optionalFeatures`. For example, if a feature does not require a signal of [user intent](privacy-security-explainer.md#user-intent). Or, if the requested session mode implies consent, such as `immersive-vr` does for `local` and `local-floor` reference spaces. 
```

Comment:
I'd suggest adding two additional concepts here:
- Performance might affect whether a feature is added implicitly
- Which features get added implicitly might vary by user agent. Don't count on anything not `required`

Also, I think I know what the last sentence means, but the phrase 'implies consent' could be interpreted as 'implies user consent' so I'm not certain. I think the intention is that session mode could imply that certain features are available.

In case it helps, here's a suggested alternative including the above concepts and clarifying the meaning around session mode (note: this isn't intended to be blocking, I'm just including it in case it's useful):

> If the UA does not recognize an entry in `XRSession.requiredFeatures` the session will not be created.
>
> Features not listed in either array may or may not be available during the session. Under some circumstances the UA may implicitly add features to `XRSessionInit.optionalFeatures`, enabling them during the session.
>
>For example, a user agent may choose to enable a feature that does not require a signal of [user intent](privacy-security-explainer.md#user-intent) and which has no adverse performance implications. This behavior is likely to vary between user agents, and developers should not depend upon the availability of features not explicitly listed as `required`.
>
> The user agent may also enable features that are implied by session type, or implied by other requested features. For example, as any immersive session must support both `local` and `local-floor` reference spaces, a user agent may add both references spaces to `XRSessionInit.optionalFeatures` when an `immersive-vr` session is created.

On the last bit, I'm curious whether this is a firm requirement. For example, consider a scenario where a developer requests an immersive session with a `local` reference space, but not `local-floor`. Suppose the user has configured their viewer height. #638 proposes that exposing actual viewer height requires [explicit consent](privacy-security-explainer.md#explicit-consent) because it is sensitive data. In this case, should the user agent:

1. Require [explicit consent](privacy-security-explainer.md#explicit-consent) because `local-floor` is implied? 
2. Add an implicit `local-floor` reference space, but use emulated floor height to avoid [explicit consent](privacy-security-explainer.md#explicit-consent)?
3. Not add `local-floor` at all, even though it's an immersive session?
4. Any of the above, at the discretion of the user agent (and if a developer needs `local-floor` they should make it `required`?)

Reply:
Good point about potential performance feedback!  And I definitely didn't make it clear enough, that it is not a choice to be left to the UA discretion about which features qualify for being automatically enabled.  Without consistency on this, we'd end up in a world of hurt.  

I also, better understand your concern now about `local-floor` technically having a different threat profile than simply `local`.  While it is a requirement that `local` and `local-floor` be *possible* in any session that supports `immersive-vr` and `immersive-ar`, it doesn't mean we necessarily want to expose user height without the UA having the opportunity to manage that risk differently.

Hopefully my latest commit has addressed both your concerns and @toji's. If so, I'd appreciate an "Approve" :)

Reply:
The table definitely helps clarify things, thanks!

To make sure I understand your above comment, Nell: Sounds like the intent is that `local-floor` is no longer one of the implicit features, but (as is currently required by the spec) would still be available to any immersive session via an emulated floor height. The only way to get a non-emulated floor height would be to explicitly request `local-floor` in either the `requiredFeatures` or `optionalFeatures` lists. Is that correct?

Reply:
This change came from my discussion with @johnpallett on Friday, so please correct me if I've misunderstood.  I was under the impression that we couldn't automatically allow a local-floor with even an emulated height because it could be used for fingerprinting.

Reply:
The concern with `local-floor` is when the viewer height is _not_ emulated. The user's height is sensitive data (i.e. 'are you a child') so would likely require explicit consent on some platforms.

In contrast, `local-floor` with _emulated_ height doesn't have this concern, and the UA can round the data to mitigate the fingerprinting risk.

@NellWaliczek If I understood on Friday the primary concern was making sure user agents were consistent in their behavior; it seems like requiring an _emulated_ `local-floor` in this scenario could achieve that consistency.

Reply:
Oh, right now I remember.  We have two competing problems to solve:

1. We definitely don't want `local-floor` to be forced to use an emulated height when a real one is available (either via sensor data or a user configuration setting).
1. We definitely don't want to have inconsistency about some hardware needing to have `local-floor` requested up front and others not.  

The intent behind this approach was simply to settle on the lowest common denominator to ensure consistency and user safety.

@toji, you on board with that?

Reply:
I'm not @toji but that makes sense to me and I agree with the approach as written

Reply:
To make sure I'm on the same page: My reading of this is that `local-floor` must be explicitly requested to be used, and `local` is the only reference space that will be implicitly available to immersive sessions. If that's correct, I'm on board with it.

Reply:
That's correct